src/pyvale/examples/basicsensorsim/ex1_scalar_sensors.py

```diff
@@ -0,0 +1,239 @@
+# list when we simulate our measurements. The error chain is the core of the
+# `pyvale` sensor simulation engine.
+
+err_chain: list[sens.IErrSimulator] = [
+    sens.ErrSysGen(sens.GenUniform(low=-10.0,high=10.0)),
+    sens.ErrRandGen(sens.GenNormal(std=5.0)),
+]
+
+sens_array.set_error_chain(err_chain)
+
+#%%
+# 3. Run a simulated experiment
+# -----------------------------
+# We have built our sensor array so now we can call `.sim_measurements()` to
+# generate simulated sensor traces. When we call this function `pyvale` will
+# calculate the ground truth (if not already complete from a previous sim), then
+# step through the error chain sampling probability distributions for our
+# errors.
+#
+# If we call `.sim_measurements()` again the process is repeated and the errors
+# are resampled. However, if we call `.get_measurements()` then we are returned
+# the previously simulated values. Throughout `pyvale` methods prefixed with
+# `get` can be expected to return previous values if they exist whereas `sim`
+# or `calc `methods will actually perform a simulation or calculation.
+measurements: np.ndarray = sens_array.sim_measurements()
+
+truth: np.ndarray = sens_array.get_truth()
+sys_errs: np.ndarray = sens_array.get_errors_systematic()
+rand_errs: np.ndarray = sens_array.get_errors_random()
+
+print(80*"-")
+print("pyvale sensor simulation model:")
+print("    measurement = truth + sysematic error + random error\n")
+
+print(f"measurements.shape = {measurements.shape} = "
+        + "(n_sensors,n_field_components,n_timesteps)\n")
+print(f"truth.shape     = {truth.shape}")
+print(f"sys_errs.shape  = {sys_errs.shape}")
+print(f"rand_errs.shape = {rand_errs.shape}")
+
+sens_print: int = 0
+comp_print: int = 0
+time_last: int = 5
+time_print = slice(measurements.shape[2]-time_last,measurements.shape[2])
+
+print(f"\nThese are the last {time_last} virtual measurements of sensor "
+        + f"{sens_print}:\n")
+
+sens.print_measurements(sens_array,sens_print,comp_print,time_print)
+print("\n"+80*"-")
+
+# %%
+#
+# .. image:: ../../../../_static/basics_ex1_term_out.png
+#    :alt: Terminal output showing the simulated measurements
+#    :width: 700px
+#    :align: center
+
+#%%
+# 4. Analyse & visualise the results
+# ----------------------------------
+# We can now visualise the sensor locations on the simulation mesh and the
+# simulated sensor traces using `pyvale` visualisation tools which are built on
+# `pyvista` for meshes and `matplotlib` for sensor traces. `pyvale` will return
+# figure and axes objects to the user allowing additional customisation using
+# `pyvista` and `matplotlib`. This also means that we need to call `.show()`
+# ourselves to display the figure as pyvale does not do this for us.
+#
+# First we create our standard 'pyvale-output' directory to save images of our
+# results to.
+output_path = Path.cwd() / "pyvale-output"
+if not output_path.is_dir():
+    output_path.mkdir(parents=True, exist_ok=True)
+
```

Comment:
Maybe the data keys should be printed here, similar to `ex3_experiment_simulator.py`?
```suggestion
print("Keys in the data dictionary:")
print(f"    {sim_data.node_vars.keys()=}")
```

Reply:
I think this is covered in the byo sim data example and the in text description of the sim_data object at the top. If we were going to add something describing the input SimData then it should probably go up the top at step 1 - the reason I didn't want a detailed description of the sim data in the basic examples was because these should just be for trying out the package to see what it does. If you then decide you want to use the package yourself you move onto the extended examples where the first one shows you how to load your own SimData from text files.